skiplang/compiler/src/skipParse.sk

```diff
@@ -111,19 +117,19 @@ fun updatePackageFiles(
     InputPackageName(name),
   ) match {
   | Some(old_pkg) -> pkgDelta(old_pkg, pkg)
-  | None() -> (pkg.srcs.map(s -> s.i0), Array[])
+  | None() -> (pkg.srcs.map(s -> (pkg.base_dir, s.i0)), Array[])
   };
-  for (src_path in modified_files) {
+  for ((pkg_base_dir, src_path) in modified_files) {
     fileDir.writeArray(
       context,
-      InputSource(name, Path.join(pkg.base_dir, src_path)),
+      InputSource(name, Path.join(pkg_base_dir, src_path)),
       Array[SKStore.StringFile(get_file_contents(src_path))],
     )
   };
-  for (src_path in removed_files) {
+  for ((pkg_base_dir, src_path) in removed_files) {
     fileDir.writeArray(
       context,
-      InputSource(name, Path.join(pkg.base_dir, src_path)),
+      InputSource(name, Path.join(pkg_base_dir, src_path)),
```

Comment:
So was the core of the issue that modified or removed files can be from a different package, in which case using `pkg.base_dir` yields the wrong file?

Reply:
Yes, in case the whole package (in this case `std` was replaced by an other implementation (in this case the bootstrap one vs the local one)) is replaced, all the files from the old package are removed, and all those of the new one added.